### PR TITLE
Fix `CopyToDevice` for `PortableHostMultiCollection` and New Test

### PR DESCRIPTION
**Pull request by AdrianoDee**
#### PR description:

This PR proposes a small fix for the `CopyToDevice` function for `PortableHostMultiCollection` and the addition of a new test to exercise the copy (in the form of an `ESProduct` based on a `PortableHostMultiCollection` to be copied to device).

#### PR validation:

`scram b runtests` runs.

### CODE REVIEW

**Comment by AdrianoDee** [comment]
type bug-fix

**Comment by AdrianoDee** [comment]
enable gpu

**Comment by AdrianoDee** [comment]
please test

**Comment by AdrianoDee** [comment]
please test

File: HeterogeneousCore/AlpakaTest/interface/AlpakaESTestRecords.h [hunk 1]
```diff
@@ -10,6 +10,8 @@ class AlpakaESTestRecordB : public edm::eventsetup::EventSetupRecordImplementati
 
 class AlpakaESTestRecordC : public edm::eventsetup::EventSetupRecordImplementation<AlpakaESTestRecordC> {};
 
+// class AlpakaESTestRecordACMulti : public edm::eventsetup::EventSetupRecordImplementation<AlpakaESTestRecordACMulti> {};
```
[the remark this answers is not in the document]
Comment (AdrianoDee) [reply HeterogeneousCore/AlpakaTest/interface/AlpakaESTestRecords.h:13]: No, it's not needed for simplicity I reused the  `AlpakaESTestRecordA` and this is a leftover. Will remove it.

File: HeterogeneousCore/AlpakaTest/src/AlpakaESTestRecords.cc [hunk 2]
```diff
@@ -5,3 +5,4 @@ EVENTSETUP_RECORD_REG(AlpakaESTestRecordA);
 EVENTSETUP_RECORD_REG(AlpakaESTestRecordB);
 EVENTSETUP_RECORD_REG(AlpakaESTestRecordC);
 EVENTSETUP_RECORD_REG(AlpakaESTestRecordD);
+// EVENTSETUP_RECORD_REG(AlpakaESTestRecordACMulti);
```
Comment (makortel) [inline HeterogeneousCore/AlpakaTest/src/AlpakaESTestRecords.cc:8]: Also this line should be removed

**Comment by makortel** [comment]
Visually looks good.

On a more general note (certainly beyond this PR), I think the tests would benefit from more verification of the ES data product contents.

**Comment by AdrianoDee** [comment]
please test

**Comment by makortel** [comment]
You need to add `ESProducerAMulti` to the list in
https://github.com/cms-sw/cmssw/blob/5515de3768cc8bd4ec0b10386ea7c641facba8b6/HeterogeneousCore/AlpakaTest/test/testAlpakaModules_cfg.py#L153-L161

**Comment by AdrianoDee** [comment]
please test

**Comment by AdrianoDee** [comment]
please test

**Comment by slava77** [comment]
@makortel @fwyzard 
is this PR OK now?
I'm asking since it's holding up an update in #45117

**Comment by makortel** [comment]
+heterogeneous

**Comment by mandrenguyen** [comment]
+1

**Pull request merged**